testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js

```diff
@@ -7604,6 +7604,58 @@ QUnit.module('Initialization', baseModuleConfig, () => {
         assert.equal(dataErrorOccurred.getCall(0).args[0].error, 'Unknown error', 'default error message');
         assert.notOk($errorRow.length, 'error row is not rendered');
     });
+
+    // T921829
+    QUnit.test('Row adding should work correctly if add button was clicked before table render', function(assert) {
+        // arrange
+        const dataGrid = createDataGrid({
+            dataSource: {
+                load: function() {
+                    const d = $.Deferred();
+                    setTimeout(() => {
+                        d.resolve([], {
+                            totalCount: 0
+                        });
+                    });
+
+                    return d;
+                }
+            },
+            columns: [{
+                dataField: 'field1',
+                fixed: true
+            }, 'field2', 'field3', 'field4', 'field5'],
+            filterRow: {
+                visible: true
+            },
+            scrolling: {
+                mode: 'virtual'
+            },
+            height: 600,
+            showBorders: true,
+            editing: {
+                allowAdding: true
+            },
+            stateStoring: {
+                enabled: true,
+                type: 'localStorage',
+                storageKey: 'storage'
+            }
+        });
```

Comment:
Simplify datagrid configuration. I think that rhe stateAtoring, filterRow options can be removed. Also, totalCount in load callback is not necessary because remoteOperaions option is not defined.

Reply:
totalCount, filterRow and virtual scrolling are not necessary, however without state storing bug is not reproduced. I will remove unnecessary options in configuration.